src/signalrclient/websocket_transport.cpp

```diff
@@ -97,7 +98,13 @@ namespace signalr
                     cts->cancel();
 
                     std::promise<void> promise;
-                    websocket_client->stop([&promise](std::exception_ptr exception)
+                    auto client = weak_websocket_client.lock();
+                    if (!client)
```

Comment:
I think receive_loop should take a completion callback and we shouldn't unreference the websocket_client until that completion callback is called.

I think the weak_ptr fine to stop cycles, but if somehow the reference is lost before the receive_loop we should raise errors as this indicates a bug.